stdlib/public/runtime/Casting.cpp

```diff
@@ -32,6 +32,11 @@
 #include "stddef.h"
 
 #include <cstring>
+#if defined(_MSC_VER)
+#include <shared_mutex>
+#else
+#include <pthread.h>
+#endif
 #include <type_traits>
```

Comment:
Please review the newly opened PR #1950. It adds read/write lock support as well is a little better factored to allow for alternate platforms to provide their own implementations if they don't have pthreads.